### ISSUE TITLE
Exception cleanup

### ISSUE THREAD

**Issue by shsms** [frequenz-floss/frequenz-client-electricity-trading-python#86]
### What's needed?

All exceptions raised by the client need to be subclasses of [ApiClientError](https://frequenz-floss.github.io/frequenz-client-base-python/latest/reference/frequenz/client/base/exception/#frequenz.client.base.exception.ApiClientError).  For non-grpc error, we could have an additional subclass called `TradingClientError`.

Then the documentation need be updated to say that users can catch all client exceptions by just catching `ApiClientError`.

**Comment by matthias-wende-frequenz** [comment]
@shsms can we close this issue?

**Comment by shsms** [comment]
I don't think this was ever done.  Maybe keep as a good first issue for future team members?
